lib/cli-engine.js

```diff
@@ -496,15 +517,19 @@ CLIEngine.prototype = {
             results.push(res);
         }
 
-        files.forEach(function(pattern) {
-            glob.sync(pattern, globOptions).forEach(executeOnFile);
+        patterns.forEach(function(pattern) {
+            if (shell.test("-f", pattern) && !ignoredPaths.contains(pattern)) {
```

Comment:
This case: https://github.com/eslint/eslint/pull/3666/files#diff-6e3261f1362e4e13d9d684e4cae1a127R398